### PR TITLE
4wif changes from upstream

### PR DESCRIPTION
**Pull request by mlyle**
Includes these betaflight commits:

2d71acf9 Increased timeouts for gigadevice clone MCUs
a8851847 Bump 4wif version number to 20001
c9f15515 Don't accept Atmel device IDs for blheli32

### CODE REVIEW

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c [hunk 1]
```diff
@@ -76,7 +76,7 @@
 // *** change to adapt Revision
 #define SERIAL_4WAY_VER_MAIN 20
 #define SERIAL_4WAY_VER_SUB_1 (uint8_t) 0
-#define SERIAL_4WAY_VER_SUB_2 (uint8_t) 00
+#define SERIAL_4WAY_VER_SUB_2 (uint8_t) 01
```
Comment (mlyle) [inline flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c:79]: This is not ideal; the version was bumped after no functional change-- a couple of months after blheli32 support was added.  Then Blhelisuite moved to depend upon the new version.

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way_avrootloader.c [hunk 2]
```diff
@@ -327,7 +327,7 @@ uint8_t BL_PageErase(ioMem_t *pMem)
     if (BL_SendCMDSetAddress(pMem)) {
         uint8_t sCMD[] = {CMD_ERASE_FLASH, 0x01};
         BL_SendBuf(sCMD, 2);
-        return (BL_GetACK((1000 / START_BIT_TIMEOUT_MS)) == brSUCCESS);
+        return (BL_GetACK((1400 / START_BIT_TIMEOUT_MS)) == brSUCCESS);
```
Comment (mlyle) [inline flight/Modules/UAVOMSPBridge/3rdparty/serial_4way_avrootloader.c:330]: These timeouts are real hacks.

**Comment by glowtape** [comment]
Seems to work. I had an odd timeout the first attempt (WriteFile function failed, semaphore timeout), but afterwards, I could flash four BLHeli_32 ESCs sequentially one after another without any incident with BLHeliSuite 32.1.

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c (hunk 1 above)
Comment (tracernz) [reply flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c:79]: Uggh.

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way_avrootloader.c (hunk 2 above)
Comment (tracernz) [reply flight/Modules/UAVOMSPBridge/3rdparty/serial_4way_avrootloader.c:330]: A fair portion of this code is... :stuck_out_tongue:

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c [hunk 3]
```diff
@@ -335,7 +335,7 @@ uint16_t _crc_xmodem_update (uint16_t crc, uint8_t data) {
         (pDeviceInfo->words[0] == 0xF850) || (pDeviceInfo->words[0] == 0xE8B1) || \
         (pDeviceInfo->words[0] == 0xE8B2))
 
-#define ARM_DEVICE_MATCH ((pDeviceInfo->words[0] == 0x9307) || (pDeviceInfo->words[0] == 0x930A) || \
+#define ARM_DEVICE_MATCH ((pDeviceInfo->words[0] == 0x1F06) || \
```
Comment (tracernz) [inline flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c:338]: Any idea what the new one is?

**Review by tracernz — APPROVED** [review]

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c (hunk 3 above)
Comment (mlyle) [reply flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c:338]: I think it's gigadevice ST-clone-crap.

File: flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c (hunk 1 above)
Comment (mlyle) [reply flight/Modules/UAVOMSPBridge/3rdparty/serial_4way.c:79]: I also like the unintentional octal here. :dagger:

**Pull request merged**